pkg/daemon/update.go

```diff
@@ -487,12 +489,38 @@ func (dn *Daemon) updateOS(oldConfig, newConfig *mcfgv1.MachineConfig) error {
 	return dn.NodeUpdaterClient.RunPivot(newConfig.Spec.OSImageURL)
 }
 
+// Log a message to the systemd journal as well as our stdout
+func (dn *Daemon) logSystem(format string, a ...interface{}) {
+	message := fmt.Sprintf(format, a...)
+	glog.Info(message)
+	// Since we're chrooted into the host rootfs with /run mounted,
+	// we can just talk to the journald socket.  Doing this as a
+	// subprocess rather than talking to journald in process since
+	// I worry about the golang library having a connection pre-chroot.
+	logger := exec.Command("logger")
+	stdin, err := logger.StdinPipe()
+	if err != nil {
+		glog.Errorf("Failed to get stdin pipe: %v", err)
+		return
+	}
+
+	go func() {
+		defer stdin.Close()
```

Comment:
This isn't problematic, but why not do the `WriteString` followed by `Close` since `defer` executes on function end?

Reply:
Because I write Go sometimes by cargo culting [bits from the Internet](https://golang.org/pkg/os/exec/#Cmd.StdinPipe) :smile:

Reply:
I think the general idea is that it's a best practice to always `defer..*close` since that way if e.g. if one later refactors a function to include an early return it still works?

Reply:
Works for me. :+1: